test/node/reducers/vis-state-merger-test.js

```diff
@@ -285,6 +287,72 @@ test('VisStateMerger.current -> mergeLayers -> toEmptyState', t => {
   t.end();
 });
 
+test('visStateMerger -> mergeLayer -> imcremental load', t => {
+  const stateToSave = cloneDeep(StateWFilesFiltersLayerColor);
+  const appStateToSave = SchemaManager.save(stateToSave);
+  const {datasets, config} = appStateToSave;
+
+  const [dataset1, dataset2] = datasets;
+  // load config first
+  const stateWithConfig = coreReducer(stateToSave, addDataToMap({config}));
+
+  t.deepEqual(
+    stateWithConfig.visState.preserveLayerOrder,
+    ['hexagon-2', 'point-0', 'geojson-1'],
+    'shoud preserve layer order'
+  );
+
+  t.deepEqual(
+    stateWithConfig.visState.layerToBeMerged.map(l => l.id),
+    ['hexagon-2', 'point-0', 'geojson-1'],
+    'should save to layerToBeMerged'
+  );
+
+  // load dataset2
+  const parsedData2 = SchemaManager.parseSavedData([dataset2]);
+  const stateWithData2 = coreReducer(stateWithConfig, addDataToMap({datasets: parsedData2}));
+  t.deepEqual(
+    stateWithData2.visState.preserveLayerOrder,
+    ['hexagon-2', 'point-0', 'geojson-1'],
+    'shoud preserve layer order'
+  );
+
+  t.deepEqual(
+    stateWithData2.visState.layers.map(l => l.id),
+    ['geojson-1'],
+    'shoud load geojeon layer'
+  );
+
+  t.deepEqual(stateWithData2.visState.layerOrder, [0], 'layerOrder should be correct');
+  t.deepEqual(
+    stateWithData2.visState.layerToBeMerged.map(l => l.id),
+    ['hexagon-2', 'point-0'],
+    'should save to layerToBeMerged'
+  );
+
+  // load dataset1
+  const parsedData1 = SchemaManager.parseSavedData([dataset1]);
+  const stateWithData1 = coreReducer(stateWithData2, addDataToMap({datasets: parsedData1}));
+  t.deepEqual(
+    stateWithData1.visState.preserveLayerOrder,
+    ['hexagon-2', 'point-0', 'geojson-1'],
+    'shoud preserve layer order'
+  );
+  t.deepEqual(
+    stateWithData1.visState.layers.map(l => l.id),
+    ['geojson-1', 'hexagon-2', 'point-0'],
+    'shoud load 2 layers'
+  );
+  t.deepEqual(stateWithData1.visState.layerOrder, [1, 2, 0], 'layerOrder should be correct');
```

Comment:
Trying to understand how the `preserveLayerOrder` works. Is the `layerOrder` the index of where the layer should go? If so, shouldn't the layerOrder be `[2, 0, 1]`? That is, `geojson-1` should be in index 2, `hexagon-2` should be in index 0, and point-0 should be at index 1?